fabfile.py

```diff
@@ -176,13 +184,52 @@ def deploy(c, target=None, commit=None):
 
 
 @task
-def rollback(c, release=None):
+def rollback(c, target=DEFAULT_TARGET, release=None):
     with Connection(c.deploy.host, user=c.deploy.user, config=c.config) as c:
         setup(c, release=release)
         update(c)
         publish(c)
         finish(c)
 
+# Please add the "target" parameter if you are adding more @task functions
+#  to allow custom targets to be used (regardless if your function itself will use it or not)
+
+
+# Source: https://realpython.com/python-command-line-arguments/#a-few-methods-for-parsing-python-command-line-arguments
+def parseArgs(systemArgs, argStart=1):
+    opts = {"":[]}
+    cmdLineLength = len(systemArgs)
+    i = argStart
+
+    while i < cmdLineLength:
+        cmdArg = systemArgs[i]
+        if cmdArg.startswith("-"):
+            cmdArg2 = systemArgs[i + 1] if (i + 1) < cmdLineLength else ""
+            if cmdArg2.startswith("-"):
+                cmdArg2 = ""
+            else:
+                i += 1
+            # cmdArg2 = cmdArg.split("=", 1)
+            # cmdArg2 = cmdArg2[1] if len(cmdArg2) == 2 else ""
+            if cmdArg not in opts:
+                opts[cmdArg] = cmdArg2
+        else:
+            opts[""].append(cmdArg)
+        i += 1
+    
+    return opts
+
+def getTarget(options, default=DEFAULT_TARGET):
+    target = options.get(TARGET_FLAG, default)
+    if target not in configs:
+        message = "\n\tTarget Configuration at {} is not a valid entry".format(TARGET_FLAG)
+        message += "\n\tInvalid Entry: " + target
+        assert target in configs, message
+    return target
+
+options = parseArgs(sys.argv)
```

Comment:
Note about variable style, I tend to follow [PEP 8](https://pep8.org/) which dictates snake_case for functions rather than camelCase. I'd prefer if we kept this consistent with the rest of the codebase.

In a perfect world we'd all `pip install black` and use [black](https://github.com/psf/black) to format all code but that may be asking a bit much. I'm not sure if black even works on Python 3.5.

Reply:
Styling Changed